### ISSUE TITLE
ImportError when using pytorch3d in Google Colab

### ISSUE THREAD

**Issue by cajoek** [facebookresearch/pytorch3d#999]
## 🐛 Bugs / Unexpected behaviors
Following the pytorch3d install instructions for Google Colab causes an `ImportError` when importing `load_obj`.

## Full notebook and outputs:
Cell 1
```python
import torch
import torchvision

!python --version
print("PyTorch version:", torch.__version__)
print("torchvision version:", torchvision.__version__)
print("# CPU:", torch.get_num_threads())
print("# GPU:", torch.cuda.device_count())
if torch.cuda.is_available():
    print("GPU type:", torch.cuda.get_device_name(0))
    print("GPU total memory:", torch.cuda.get_device_properties(0).total_memory * 1e-9, "GB")
```

```
Python 3.7.12
PyTorch version: 1.10.0+cu111
torchvision version: 0.11.1+cu111
# CPU: 1
# GPU: 1
GPU type: Tesla K80
GPU total memory: 11.996954624 GB
```
Cell 2 (Copied from  pytorch3d/INSTALL.md)
```python
import sys
import torch
version_str="".join([
    f"py3{sys.version_info.minor}_cu",
    torch.version.cuda.replace(".",""),
    f"_pyt{torch.__version__[0:5:2]}"
])
!pip install pytorch3d -f https://dl.fbaipublicfiles.com/pytorch3d/packaging/wheels/{version_str}/download.html
```

```
Looking in links: https://dl.fbaipublicfiles.com/pytorch3d/packaging/wheels/py37_cu111_pyt11./download.html
Collecting pytorch3d
  Downloading pytorch3d-0.3.0-cp37-cp37m-manylinux1_x86_64.whl (30.0 MB)
     |████████████████████████████████| 30.0 MB 1.8 MB/s 
Requirement already satisfied: torchvision>=0.4 in /usr/local/lib/python3.7/dist-packages (from pytorch3d) (0.11.1+cu111)
Collecting fvcore
  Downloading fvcore-0.1.5.post20211023.tar.gz (49 kB)
     |████████████████████████████████| 49 kB 6.1 MB/s 
Requirement already satisfied: torch==1.10.0 in /usr/local/lib/python3.7/dist-packages (from torchvision>=0.4->pytorch3d) (1.10.0+cu111)
Requirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from torchvision>=0.4->pytorch3d) (1.19.5)
Requirement already satisfied: pillow!=8.3.0,>=5.3.0 in /usr/local/lib/python3.7/dist-packages (from torchvision>=0.4->pytorch3d) (7.1.2)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.7/dist-packages (from torch==1.10.0->torchvision>=0.4->pytorch3d) (3.10.0.2)
Collecting yacs>=0.1.6
  Downloading yacs-0.1.8-py3-none-any.whl (14 kB)
Collecting pyyaml>=5.1
  Downloading PyYAML-6.0-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (596 kB)
     |████████████████████████████████| 596 kB 48.6 MB/s 
Requirement already satisfied: tqdm in /usr/local/lib/python3.7/dist-packages (from fvcore->pytorch3d) (4.62.3)
Requirement already satisfied: termcolor>=1.1 in /usr/local/lib/python3.7/dist-packages (from fvcore->pytorch3d) (1.1.0)
Requirement already satisfied: tabulate in /usr/local/lib/python3.7/dist-packages (from fvcore->pytorch3d) (0.8.9)
Collecting iopath>=0.1.7
  Downloading iopath-0.1.9-py3-none-any.whl (27 kB)
Collecting portalocker
  Downloading portalocker-2.3.2-py2.py3-none-any.whl (15 kB)
Building wheels for collected packages: fvcore
  Building wheel for fvcore (setup.py) ... done
  Created wheel for fvcore: filename=fvcore-0.1.5.post20211023-py3-none-any.whl size=60947 sha256=70a2d46c397662b08b8ed06c12548b4328bd39179f5d1fc18a042232c2695838
  Stored in directory: /root/.cache/pip/wheels/16/98/fc/252d62cab6263c719120e06b28f3378af59b52ce7a20e81852
Successfully built fvcore
Installing collected packages: pyyaml, portalocker, yacs, iopath, fvcore, pytorch3d
  Attempting uninstall: pyyaml
    Found existing installation: PyYAML 3.13
    Uninstalling PyYAML-3.13:
      Successfully uninstalled PyYAML-3.13
Successfully installed fvcore-0.1.5.post20211023 iopath-0.1.9 portalocker-2.3.2 pytorch3d-0.3.0 pyyaml-6.0 yacs-0.1.8
```
Cell 3 (`ImportError`)
```python
import pytorch3d
from pytorch3d.io import load_obj
```

```
---------------------------------------------------------------------------

ImportError                               Traceback (most recent call last)

<ipython-input-7-dfdca11852fa> in <module>()
      1 import pytorch3d
----> 2 from pytorch3d.io import load_obj

3 frames

/usr/local/lib/python3.7/dist-packages/pytorch3d/io/__init__.py in <module>()
      2 
      3 
----> 4 from .obj_io import load_obj, load_objs_as_meshes, save_obj
      5 from .ply_io import load_ply, save_ply
      6 

/usr/local/lib/python3.7/dist-packages/pytorch3d/io/obj_io.py in <module>()
     12 from pytorch3d.io.mtl_io import load_mtl, make_mesh_texture_atlas
     13 from pytorch3d.io.utils import _check_faces_indices, _make_tensor, _open_file
---> 14 from pytorch3d.renderer import TexturesAtlas, TexturesUV
     15 from pytorch3d.structures import Meshes, join_meshes_as_batch
     16 

/usr/local/lib/python3.7/dist-packages/pytorch3d/renderer/__init__.py in <module>()
      1 # Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
      2 
----> 3 from .blending import (
      4     BlendParams,
      5     hard_rgb_blend,

/usr/local/lib/python3.7/dist-packages/pytorch3d/renderer/blending.py in <module>()
      7 
      8 # pyre-fixme[21]: Could not find name `_C` in `pytorch3d`.
----> 9 from pytorch3d import _C
     10 
     11 

ImportError: /usr/local/lib/python3.7/dist-packages/pytorch3d/_C.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZNK2at6Tensor7is_cudaEv


---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.
---------------------------------------------------------------------------
```
Is the wrong version of pytorch3d being installed?

**Comment by cajoek** [comment]
Looks like version 0.3 is being installed rather than 0.6

**Comment by bottler** [comment]
You need to change cell 2 to 
```
import sys
import torch
pyt_version_str=torch.__version__.split("+")[0].replace(".", "")
version_str="".join([
    f"py3{sys.version_info.minor}_cu",
    torch.version.cuda.replace(".",""),
    f"_pyt{pyt_version_str}"
])
!pip install pytorch3d -f https://dl.fbaipublicfiles.com/pytorch3d/packaging/wheels/{version_str}/download.html
```

This is because we now support PyTorch 1.10.0, which you have. The previous code assumed that the PyTorch version number was major.minor.patch with each of major, minor and patch being a single digit. I updated the code in the tutorials at the `stable` tag (on the `v0.6` release branch) straight after the release of 0.6.1 last week, in 3fc3ea5, and the main branch (including INSTALL.md) was fixed in 1152a93.

**Comment by cajoek** [comment]
Thanks it worked!
